flink-staging/flink-streaming/flink-streaming-core/src/main/java/org/apache/flink/streaming/runtime/io/StreamingAbstractRecordReader.java

```diff
@@ -57,6 +58,12 @@
 
 	private final BarrierBuffer barrierBuffer;
 
+	/**
+	 * Counters for the number of bytes read and records processed.
+	 */
+	private LongCounter numRecordsRead = null;
```

Comment:
`null` initializations are actually redundant. They still get executed (for OpenJDK javac and Oracle javac), so it is actually overhead for no reason.